worlds/metroidprime/data/Tricks.py

```diff
@@ -0,0 +1,824 @@
+        TrickDifficulty.Hard,
+        lambda world, state: can_space_jump(world, state)
+        and can_morph_ball(world, state),
+    )
+
+    magma_pool_scan_dash: TrickInfo = TrickInfo(
+        "Cross Magma Pool Suitless",
+        "Cross magma pool using a scan dash on the crate items",
+        TrickDifficulty.Medium,
+        lambda world, state: can_space_jump(world, state) and can_scan(world, state),
+    )
+    magma_pool_debris_jump: TrickInfo = TrickInfo(
+        "Cross Magma Pool With SJB and Gravity",
+        "Use the space jump boots to jump off debris to cross the pool",
+        TrickDifficulty.Easy,
+        lambda world, state: can_space_jump(world, state)
+        and can_move_underwater(world, state)
+        and has_energy_tanks(world, state, 2),
+    )
+    magma_pool_item_debris_jump: TrickInfo = TrickInfo(
+        "Magma Pool Debris Jump",
+        "Use the space jump boots to jump off debris to cross the pool",
+        TrickDifficulty.Easy,
+        lambda world, state: can_space_jump(world, state)
+        and can_move_underwater(world, state)
+        and has_energy_tanks(world, state, 2)
+        and can_power_bomb(world, state),
+    )
+    magma_pool_item_scan_dash: TrickInfo = TrickInfo(
+        "Magma Pool Item No Grapple",
+        "Use the scan dash and a power bomb to get the item in the magma pool",
+        TrickDifficulty.Medium,
+        lambda world, state: can_space_jump(world, state)
+        and can_scan(world, state)
+        and can_power_bomb(world, state),
+    )
+    magma_pool_item_infinite_speed: TrickInfo = TrickInfo(
+        "Magma Pool Item Infinite Speed",
+        "Use infinite speed to get the item in the magma pool. Stay in morph ball and navigate to a save station or elevator afterward",
+        TrickDifficulty.Medium,
+        rule_func=lambda world, state: can_infinite_speed(world, state)
+        and can_bomb(world, state),
+    )
+
+    arboretum_scan_gate_skip: TrickInfo = TrickInfo(
+        "Arboretum Scan Gate Skip",
+        "Skip the gate in the Arboretum by double bomb jumping",
+        TrickDifficulty.Easy,
+        can_bomb,
+    )
+
+    gathering_hall_without_space_jump: TrickInfo = TrickInfo(
+        "Gathering Hall Without Space Jump",
+        "Double bomb jump from the side platform to the grate where the item is",
+        TrickDifficulty.Easy,
+        can_bomb,
+    )
+
+    watery_hall_no_gravity: TrickInfo = TrickInfo(
+        "Watery Hall No Gravity",
+        "Reach the Watery Hall Underwater Item without Gravity Suit by using a slope jump",
+        TrickDifficulty.Easy,
+        can_space_jump,
+    )
+    watery_hall_no_gravity_no_space_jump: TrickInfo = TrickInfo(
+        "Watery Hall No Gravity No Space Jump",
+        "Reach the Watery Hall Underwater Item without Gravity Suit or Space Jump by using a slope jump",
+        TrickDifficulty.Medium,
+        lambda world, state: can_move_underwater(world, state) == False

```

Comment:
The client actually exposes a way to disable the gravity suit, so you won't ever actually get locked out of checks like this. I can remove this since you always start with the power suit and can switch back to that temporarily